packages/playwright/src/worker/testInfo.ts

```diff
@@ -346,6 +346,16 @@ export class TestInfoImpl implements TestInfo {
       serialized.stack = `${(error as Error).name}: ${(error as Error).message}\n${stringifyStackFrames(step.boxedStack).join('\n')}`;
     this.errors.push(serialized);
     this._tracing.appendForError(serialized);
+    // We don't know if this occurred in test or hooks
+    this._checkForFloatingPromises('the test');
```

Comment:
Why specifically here? Sounds like a random place in the middle of a test. We should do it at the end to avoid false positives.